### PR TITLE
Fix column uppercasing

### LINKED ISSUE #399: Column Level Privileges are capitalized when applied to DB instance
Please raise issues via the [new interface](https://github.com/ansible/ansible/issues/new/choose)
Currently I use the mysql-roles module to manage all app roles in our db instances. I have some column level privileges to a larger table that I don't want to grant access to the whole table to some roles. I have been receiving access denied errors now because when the mysql-roles module applied the grants it capitalized the column names.

In MySQL's documentation I have found that it states that column level privileges are not case sensitive but after speaking to an oracle support rep he has confirmed that they are.

I am confused on why the module when inputting the grants for column level privileges is uppercasing the column name and not leaving it the way I have it in my playbook example follows:

What I have in my playbook:
'database.table': 'SELECT(`app_type`, `rec_id`)'

What MySQL is returning after application:
GRANT SELECT (`APP_TYPE`, `REC_ID`) ON `database`.`table` TO `role`@`%`

### PR DESCRIPTION
**Pull request by kmarse**
##### SUMMARY
This pull request is to add logic to remove uppercasing column level privileges as they are case sensitive per MySQL databases.

This fixes #399 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/modules/mysql_role.py
plugins/modules/mysql_user.py
plugins/module_utils/user.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Andersson007** [comment]
@kmarse hello, thanks for the PR!
Could you please:
- fix sanity [errors](https://github.com/ansible-collections/community.mysql/actions/runs/6102757359/job/16561888697?pr=569)
- add some integration tests for user and role to tests/integration/targets/*
- add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment)

if you have any questions, please ask

**Comment by laurent-indermuehle** [comment]
@Andersson007 I feel like we should change the default in 4.0 to true ! We could announce it like you did in #408. Are you ok?

@kmarse, hope you're doing well with everything on your plate recently. I've got the integration tests completed for the changes in #399 - they're all passing. I want to get this merged as soon as we can as #572  is blocked by it. When you get a chance, would you mind taking a look at either granting me push access to your fork's `main` branch so I can add the test files? Otherwise I can open a PR there but it's less easy. I appreciate you taking the time. 

If you think you won't have bandwidth for it in the next day or two though, feel free to let me know and I can open a PR on my end to move things along. Just wanted to check in first since it's your code. Thanks so much!

**Comment by kmarse** [comment]
@laurent-indermuehle I have added you to my fork am just waiting your acceptance now. Sorry I am pretty busy on other projects and don't have the bandwidth to help a lot on this.

**Comment by laurent-indermuehle** [comment]
@kmarse thanks for the invitation, I accepted it. I'll try to push my tests tomorrow.

**Comment by Andersson007** [comment]
> @Andersson007 I feel like we should change the default in 4.0 to true ! We could announce it like you did in https://github.com/ansible-collections/community.mysql/pull/408. Are you ok?

@laurent-indermuehle yeah, i'm fine with that, thanks

**Comment by laurent-indermuehle** [comment]
I don't get why this PR has been closed.
Maybe it's because kmarse was on 3.7.0 and our main on 3.7.2. I rebased and pushed again.

File: changelogs/569_fix_column_uppercasing.yml
Comment (Andersson007) [inline changelogs/569_fix_column_uppercasing.yml]:
```suggestion
    changed to ``true`` in community.mysql 4.0.0. If your playbook expected the
```

File: changelogs/569_fix_column_uppercasing.yml
Comment (Andersson007) [inline changelogs/569_fix_column_uppercasing.yml]:
```suggestion
    changed to ``true`` in community.mysql 4.0.0. If your playbook expected the
```

File: plugins/modules/mysql_role.py
Comment (Andersson007) [inline plugins/modules/mysql_role.py]:
```suggestion
        to C(true) in most cases.
```

File: plugins/modules/mysql_role.py
Comment (Andersson007) [inline plugins/modules/mysql_role.py]:
```suggestion
        column_case_sensitive=dict(type='bool', default=False),
```

**Review by Andersson007 — COMMENTED** [review]
@laurent-indermuehle thanks for continuing the work!
How about:
- Adding yourself and @kmarse to the `authors:` ?
- Adding a warning when the field is not set?

File: tests/unit/plugins/module_utils/test_mysql_user.py
Comment (Andersson007) [inline tests/unit/plugins/module_utils/test_mysql_user.py]:
```suggestion
    privileges_unpack,
```
it's a general recommendation I saw in several Python books

**Comment by laurent-indermuehle** [comment]
> @laurent-indermuehle thanks for continuing the work! How about:
> 
> * Adding yourself and @kmarse to the `authors:` ?
> * Adding a warning when the field is not set?

@Andersson007  module.params['column_case_sensitive'] seems to be always sets. 

Then I made a simple test "if not column_case_sensitive". Hope  that's ok: https://github.com/ansible-collections/community.mysql/pull/569/commits/baf947da4b7115da15a664d39a972b4546c34099

File: plugins/modules/mysql_role.py
Comment (Andersson007) [inline plugins/modules/mysql_role.py]:
```suggestion
    description:
      - The default is C(false).
      - When C(true), the module will not uppercase the field in the privileges.
      - When C(false), the field names will be upper-cased. This was the default before this
        feature was introduced but since MySQL/MariaDB is case sensitive you should set this
        to C(true) in most cases.
```

File: plugins/modules/mysql_role.py
Comment (Andersson007) [inline plugins/modules/mysql_role.py]:
```suggestion
        module.warn("Option column_case_sensitive is not provided. "
                    "The default is now false, so the column's name will be uppercased. "
                    "The default will be changed to true in community.mysql 4.0.0.")
```

File: plugins/modules/mysql_user.py
Comment (Andersson007) [inline plugins/modules/mysql_user.py]:
```suggestion
    description:
      - The default is C(false).
      - When C(true), the module will not uppercase the field names in the privileges.
      - When C(false), the field names will be upper-cased. This is the default
      - This feature was introduced because MySQL 8 and above uses case sensitive
        fields names in privileges.
```

File: plugins/modules/mysql_user.py
Comment (Andersson007) [inline plugins/modules/mysql_user.py]:
```suggestion
        module.warn("Option column_case_sensitive is not provided. "
                    "The default is now false, so the column's name will be uppercased. "
                    "The default will be changed to true in community.mysql 4.0.0.")
```

File: changelogs/569_fix_column_uppercasing.yml
Comment (Andersson007) [inline changelogs/569_fix_column_uppercasing.yml]:
```suggestion
minor_changes:

  - mysql_user - add ``column_case_sensitive`` option to prevent field names
    from being uppercased
    (https://github.com/ansible-collections/community.mysql/pull/569).
  - mysql_role - add ``column_case_sensitive`` option to prevent field names
    from being uppercased
    (https://github.com/ansible-collections/community.mysql/pull/569).

major_changes:
```

**Review by Andersson007 — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@kmarse @laurent-indermuehle thanks a lot for the contribution!

**Pull request merged**
